packages/react/src/internal/components/InputLabel.tsx

```diff
@@ -38,37 +38,62 @@ const InputLabel: React.FC<React.PropsWithChildren<Props>> = ({
   as = 'label',
   className,
   ...props
-}) => {
+}: Props) {
   return (
-    <VisuallyHidden
-      isVisible={!visuallyHidden}
-      as={
-        as as 'label' /* This assertion is clearly wrong, but it's the only way TS will allow the htmlFor prop to be possibly defined */
-      }
+    <StyledLabel
+      as={as}
+      data-control-disabled={disabled ? '' : undefined}
+      data-visually-hidden={visuallyHidden ? '' : undefined}
       htmlFor={htmlFor}
       id={id}
       className={className}
-      sx={{
-        fontWeight: 'bold',
-        fontSize: 1,
-        display: 'block',
-        color: disabled ? 'fg.muted' : 'fg.default',
-        cursor: disabled ? 'not-allowed' : 'pointer',
-        alignSelf: 'flex-start',
-        ...sx,
-      }}
+      sx={sx}
       {...props}
     >
       {required || requiredText ? (
-        <Box display="flex" as="span">
-          <Box mr={1}>{children}</Box>
+        <StyledRequiredText>
+          <div>{children}</div>
```

Comment:
nit, but since this in inside a span, could it also be `<span>`?

```suggestion
          <span>{children}</span>
```

Reply:
@jonrohan I think I got `span` for `StyledRequiredText` which should map to `<Box as="span"` but for the inner wrapper around `children` I think `Box` will default to `div`, or is that not the case? 👀

Reply:
Think that's the case. And what you have matches existing. I get old school ick vibes having a block element inside an inline. I don't think it matters anymore plus these are resolving to flex anyways so it doesn't really matter 🤷🏻

Reply:
@jonrohan ah, sorry I totally misunderstood you earlier. Happy to update it to be span! What you said makes a lot of sense to me 👍